src/py42/services/_connection.py

```diff
@@ -244,6 +245,14 @@ def _prepare_request(
         _print_request(method, url, params=params, data=data)
         return self._session.prepare_request(request)
 
+    def _set_content_header(self, data):
```

Comment:
This isn't thread-safe -- if we were hitting this code in one of our batch cli processes for example, we could be removing/adding  a header right before a request that was supposed to have the opposite happen was sent.